core/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -524,4 +524,137 @@ public void testParentObjectMapperAreNested() throws Exception {
         assertFalse(objectMapper.parentObjectMapperAreNested(mapperService));
     }
 
+    public void testLimitNestedDocsDefaultSettings() throws Exception{
+        MapperService mapperService = createIndex("test1", Settings.builder().build()).mapperService();
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type").startObject("properties")
+            .startObject("nested1").field("type", "nested").endObject()
+            .endObject().endObject().endObject().string();
+        DocumentMapper docMapper = mapperService.documentMapperParser().parse("type", new CompressedXContent(mapping));
+
+        // parsing a doc with 3 nested objects succeeds
```

Comment:
I think the case where #nested_docs < limit is already covered by the later tests, could you change this to that the the document has more than the default number of nested docs (simple loop) and gets rejected? This way we can test the default value works if no explicit setting is there. I'd also check if that unit test then takes a lot more time (which I doubt, gut feeling check is okay here I think)